src/main/js/dynamic/loaded/users/PermissionTable.js

```diff
@@ -147,13 +156,30 @@ class PermissionTable extends Component {
     }
 
     createContextOptions() {
-        return [{
-            label: CONTEXT_TYPE.DISTRIBUTION,
-            value: CONTEXT_TYPE.DISTRIBUTION
-        }, {
-            label: CONTEXT_TYPE.GLOBAL,
-            value: CONTEXT_TYPE.GLOBAL
-        }];
+        const { permissionsData } = this.state;
+        const { descriptors } = this.props;
+
+        const availableContexts = [];
+        if (permissionsData && permissionsData.descriptorName) {
+            descriptors.forEach((descriptor) => {
+                if (descriptor.label === permissionsData.descriptorName) {
+                    availableContexts.push(descriptor.context);
+                }
+            });
+
+            if (permissionsData.context && !availableContexts.includes(permissionsData.context)) {
+                this.setState({
+                    permissionsData: {
+                        descriptorName: permissionsData.descriptorName
+                    }
+                });
+            }
+        }
```

Comment:
If the user changes descriptors, clear the context if it is no longer relevant to the new descriptor.